source/extensions/filters/network/common/redis/fault_impl.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "envoy/api/api.h"
+#include "envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.pb.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/common/redis/fault.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class Fault {
```

Comment:
Is this class even used outside of FaultManagerImpl?  If not, do we need to be exposing this?

Reply:
Correct, we don't need to expose. I'll make it a member class with a static method on the fault manager for creating the test faults we need for testing the command splitter.